### DISCUSSION TITLE
Change layout template path completely?

### DISCUSSION THREAD

**Discussion by ev3rlost** [livewire/livewire#7835] [1. Help] [answered]
Is it possible to change the path of a layout completely - meaning outside of the `resources/views` folder?

Or is it rigidly bound to Laravels `resources/views/components/layouts/` folder?

I'm fiddling around with the modular monolith approach of a Laravel structure following this course:

https://laracasts.com/series/modular-laravel

And I'm curious if there's a way to change the layouts path, as all my Livewire components live in the `modules` folder in Laravels root directory. I've managed to make it work with Livewire, it's just the layout part thats giving me headaches.

Thanks for any advice!

**Comment by joshhanley** [comment]
@ev3rlost If you publish the Livewire config, you should be able to change it there.

**Reply by ev3rlost** [reply]
That would affect the whole app. The `modules` folder is meant for the admin pages, and I still need the standard Laravel structure for the frontend (website and other).

Looking at `'layout' => 'components.layouts.app',` - you're actually still bound to the `resources/views` folder if I'm not mistaken?

I was wondering if there was another way, using path helpers (`base_path()`) or something?

**Reply by ev3rlost** [reply, marked as answer]
Ok, I think I've made it to work as expected. (kind of)

Following the series posted above, my Laravel structure looks like this:

```
[laravel] ¬
    [app]
    [bootstrap]
    [config]
    [database]
    [modules] ¬
        [Layouts] ¬
            auth.blade.php
            admin.blade.php
        [Auth] ¬
            [Livewire] ¬
                Login.php
            [Providers] ¬
                AuthServiceProvider.php
            [Views] ¬
                [Livewire] ¬
                    login.blade.php
        [User] ¬
            [Livewire] ¬
                Users.php
            [Providers] ¬
                UserServiceProvider.php
            [Views] ¬
                [Livewire] ¬
                    users.blade.php
        GlobalServiceProvider.php
    [node_modules]
    [public]
    etc...
```

Each module (Auth, User, etc...) has its own service provider (AuthServiceProvider.php etc...) 

I had to add the following to each of the modules service provider (inside the boot method):

```
$this->loadViewsFrom(__DIR__ . '/../../Layouts', 'layout');
```

I can now use full-page components and giving them the appropriate layout using the Layout attribute
```
#[Layout('layout::admin')]
```

The `GlobalServiceProvider.php` registers all of the modules providers in the boot method. I was hoping to make it work there, but it doesn't seem to work.

If anyone find a way to make it work inside the GlobalServiceProvider.php let me know!
